tests/loggen/tests/test_loggen_filereader.c

```diff
@@ -0,0 +1,262 @@
+      .expected_pid = "-",
+      .expected_msgid = "-",
+      .expected_sdata = "[sdata_test=\"1\"]",
+      .expected_message = "test message for loggen"
+    },
+    {
+      .skip_tokens = 0,
+      .line = "<01>32 2016-04-15T11:00:06+02:00 - - - - [sdata_test=\"1\"] ﻿",
+      .expected_pri = "01",
+      .expected_ver = "32",
+      .expected_time_stamp = "2016-04-15T11:00:06+02:00",
+      .expected_host = "-",
+      .expected_app = "-",
+      .expected_pid = "-",
+      .expected_msgid = "-",
+      .expected_sdata = "[sdata_test=\"1\"]",
+      .expected_message = ""
+    },
+    {
+      .skip_tokens = 0,
+      .line = "<01>32 2016-04-15T11:00:06+02:00 - - - - -",
+      .expected_pri = "01",
+      .expected_ver = "32",
+      .expected_time_stamp = "2016-04-15T11:00:06+02:00",
+      .expected_host = "-",
+      .expected_app = "-",
+      .expected_pid = "-",
+      .expected_msgid = "-",
+      .expected_sdata = "-",
+      .expected_message = ""
+    },
+    {
+      .skip_tokens = 0,
+      .line = "<123>3210 2016-04-15T11:00:06+02:00 - - - - -",
+      .expected_pri = "123",
+      .expected_ver = "32", /* ver shall be truncated to 2 char long */
+      .expected_time_stamp = "2016-04-15T11:00:06+02:00",
+      .expected_host = "-",
+      .expected_app = "-",
+      .expected_pid = "-",
+      .expected_msgid = "-",
+      .expected_sdata = "-",
+      .expected_message = ""
+    },
+    /* RFC3164 log format */
+    {
+      .skip_tokens = 1,
+      .line = "499 <134>Feb 04 16:22:31 zts-win001 Microsoft-Windows-Winlogon[3720]: : Application Microsoft-Windows-Winlogon: [Information] Windows license validated. (EventID 4101)",
+      .expected_pri = "134",
+      .expected_ver = "",
+      .expected_time_stamp = "Feb 04 16:22:31",
+      .expected_host = "zts-win001",
+      .expected_app = "Microsoft-Windows-Winlogon",
+      .expected_pid = "3720",
+      .expected_msgid = "",
+      .expected_sdata = "",
+      .expected_message = "Application Microsoft-Windows-Winlogon: [Information] Windows license validated. (EventID 4101)"
+    },
+    {
+      .skip_tokens = 0,
+      .line = "<134>Jul 27 14:55:12 syslogng-win7 Microsoft-Windows-Service_Control_Manager[3720]: : System Microsoft-Windows-Service Control Manager: [Information] The Multimedia Class Scheduler service entered the running state. (EventID 7036)",
+      .expected_pri = "134",
+      .expected_ver = "",
+      .expected_time_stamp = "Jul 27 14:55:12",
+      .expected_host = "syslogng-win7",
+      .expected_app = "Microsoft-Windows-Service_Control_Manager",
+      .expected_pid = "3720",
+      .expected_msgid = "",
+      .expected_sdata = "",
+      .expected_message = "System Microsoft-Windows-Service Control Manager: [Information] The Multimedia Class Scheduler service entered the running state. (EventID 7036)"
+    },
+    {
+      .skip_tokens = 0,
+      .line = "<34>Oct 11 22:14:15 mymachine su[123]: : 'su root' failed for lonvick on /dev/pts/8",
```

Comment:
Why do you have 2 colons after the process name (and pid) in all RFC3164 tests? Is this a valid RFC3164 message?

What about the examples below? They are not parsed correctly.

```
<13>Feb  5 17:32:18 10.0.0.99 Use the BFG!
<13>Feb 5 17:32:18 10.0.0.99 testapp: Use the BFG!
<34>Oct 11 22:14:15 mymachine su[123]: 'su root' failed for lonvick on /dev/pts/8
```

Reply:
the 2 colons line format is used by our testers. I want to be sure the loggen is able to parse the same messages as before.
unfortunatley the RFC3164 is hard to parse due to many optional fields. I rewrite my RFC3164 perser function no it supports all (I hope so) format of RFC3164. Regarding your three example logs: I added them to the unit test to be sure we can parse it now.